### PR TITLE
TT-13671: add Telemetry docs for Cloud

### PR DESCRIPTION
**Pull request by umit**
### For internal users - Please add a Jira DX PR ticket to the subject!

---
[TT-13671](https://tyktech.atlassian.net/browse/TT-13671)

### Preview Link

<br>

[Preview](https://deploy-preview-5816--tyk-docs.netlify.app)

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [ ] I have reviewed the suggestions made by our AI (PR Agent) and updated them accordingly (spelling errors, rephrasing, etc.)
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labeled** the PR

___

### **PR Type**
Documentation

___

### **Description**
- Introduced new documentation for Telemetry in Tyk Cloud, explaining its purpose and supported providers.
- Added a comprehensive guide for setting up Telemetry connections, including provider-specific configurations for Datadog, Dynatrace, New Relic, Elastic, and custom setups.
- Updated the menu structure to include links to the new Telemetry documentation pages.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>telemetry.md</strong><dd><code>Add introductory documentation for Telemetry in Tyk Cloud.</code></dd></summary>
<hr>

tyk-docs/content/tyk-cloud/telemetry.md

<li>Added a new page introducing Telemetry in Tyk Cloud.<br> <li> Described the purpose and benefits of Telemetry for monitoring APIs.<br> <li> Listed supported Telemetry providers with links to their setup <br>instructions.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5816/files#diff-f1a37ec1f2793569acc57b916f72cd188f4272e34dbc51e37f84d46affbb005c">+23/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>enable-telemetry.md</strong><dd><code>Add detailed setup guide for Telemetry connections.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md

<li>Added detailed instructions for setting up Telemetry connections.<br> <li> Included provider-specific configuration steps for Datadog, Dynatrace, <br>New Relic, Elastic, and custom providers.<br> <li> Explained optional settings like tags and data filtering.<br> <li> Provided examples and images for better clarity.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5816/files#diff-1e4306a53aee483a70c6b6ecdd8b959005a9cb26cda6a2571263c822c33bb49a">+121/-0</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Update menu to include Telemetry documentation links.</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/data/menu.yaml

<li>Updated the menu structure to include a new "Telemetry" section.<br> <li> Added links to the "Overview" and "Setting Up Your Telemetry <br>Connection" pages.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5816/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+12/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

[TT-13671]: https://tyktech.atlassian.net/browse/TT-13671?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tyk-docs/content/tyk-cloud/telemetry.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+title: "Telemetry"
+date: 2020-04-28
+weight: 1
+tags: ["Tyk Cloud", "Configuration", "Telemetry"]
+description: "Telemetry"
+---
+
+## Introduction
+
+Telemetry in Tyk Cloud helps you monitor and understand how your APIs are performing. Think of it as a health monitoring system for your APIs - it collects data about response times, error rates, and other vital metrics that help you ensure everything is running smoothly.
```
Comment (excieve) [inline tyk-docs/content/tyk-cloud/telemetry.md]: Monitoring isn't a major part of this currently. Tyk only supports tracing atm, while some metrics may be derived from it. So I would instead highlight tracing here.

File: tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  Provider Site: us5.datadoghq.com
+  API Key: your-datadog-api-key
+  ```
+
+{{< img src="/img/cloud/telemetry-datadog.png" alt="Tyk Cloud Telemetry Datadog" >}}
+
+#### Dynatrace
+
+- Provider Endpoint: Your New Relic HTTP endpoint
+- API Token: Your New Relic license key
```
Comment (excieve) [inline tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md]: This section is about Dynatrace, but it mentions New Relic?

File: tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  Provider Endpoint: https://<YOUR-ENVIRONMENT-STRING>.live.dynatrace.com/api/v2/otlp
+  API Token: your-dynatrace-token
+  ```
+
+{{< img src="/img/cloud/telemetry-dynatrace.png" alt="Tyk Cloud Telemetry Dynatrace" >}}
+
+#### New Relic
+
+- Provider Endpoint: Your New Relic HTTP endpoint
+- API Token: Your New Relic license key
```
Comment (excieve) [inline tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md]: I don't think it's called a `license key`?

File: tyk-docs/content/tyk-cloud/telemetry.md (hunk 1 above)
Comment (excieve) [reply tyk-docs/content/tyk-cloud/telemetry.md]: Could you please also add a link to the [underlying Tyk Gateway feature](https://tyk.io/docs/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview/), which provides a bit more details about the functionality and its usage?

**Review by excieve — CHANGES_REQUESTED** [review]
Thanks @umit! There are a couple of comments. Also, could you please add a paragraph somewhere in the overview section that this is an add-on feature that can be enabled by contacting Tyk.

**Comment by excieve** [comment]
@mitjaziv Could you please give it a quick review too?

File: tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md (hunk 3 above)
Comment (umit) [reply tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md]: I checked New Relic documentation in this link: https://docs.newrelic.com/docs/data-apis/ingest-apis/telemetry-sdks-report-custom-telemetry-data/#requirements.

They are called `license key`.

File: tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md (hunk 2 above)
Comment (umit) [reply tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md]: @excieve it's fixed.

File: tyk-docs/content/tyk-cloud/telemetry.md (hunk 1 above)
Comment (umit) [reply tyk-docs/content/tyk-cloud/telemetry.md]: @excieve I've updated description and added [Tyk Gateway feature](https://tyk.io/docs/product-stack/tyk-gateway/advanced-configurations/distributed-tracing/open-telemetry/open-telemetry-overview/) link.

File: tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md (hunk 3 above)
Comment (excieve) [reply tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md]: @umit Oh, I see. They call the header `api-key`, but it's actually their license key as mentioned here: https://docs.newrelic.com/docs/opentelemetry/best-practices/opentelemetry-otlp/#api-key

**Review by excieve — APPROVED** [review]
LGTM, thanks @umit!

File: tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  Secret Token: your-elastic-secret-token
+  ```
+
+{{< img src="/img/cloud/telemetry-elastic.png" alt="Tyk Cloud Telemetry Elastic" >}}
+
+#### Custom
+
+For when you need to connect to a different monitoring system:
+
+- Exporter: Choose gRPC (currently the supported protocol)
```
Comment (mitjaziv) [inline tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md]: Both HTTP and gRPC are supported with Custom provider.

File: tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+- Provider Endpoint: Your monitoring system's URL
```
Comment (mitjaziv) [inline tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md]: `Your monitoring system URL`

**Review by mitjaziv — APPROVED** [review]
Looks good to me.

I would suggest just maybe to add info, if general Telemetry settings are changed, that cloud data planes need to be redeployed for changes to take effect.

File: tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md (hunk 5 above)
Comment (umit) [reply tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md]: @mitjaziv done.

File: tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md (hunk 4 above)
Comment (umit) [reply tyk-docs/content/tyk-cloud/telemetry/enable-telemetry.md]: @mitjaziv It's done.

**Comment by umit** [comment]
> Looks good to me.
> 
> I would suggest just maybe to add info, if general Telemetry settings are changed, that cloud data planes need to be redeployed for changes to take effect.

@mitjaziv I've added this information in `Telemetry Export in Cloud Data Plane Deployments` section.

**Review by sharadregoti — APPROVED** [review]

**Pull request merged**